pkg/build/registry/build/strategy.go

```diff
@@ -109,27 +110,51 @@ type detailsStrategy struct {
 }
 
 // Prepares a build for update by only allowing an update to build details.
-// For now, this is the Spec.Revision field
+// Build details currently consists of: Spec.Revision, Status.Reason, and
+// Status.Message, all of which are updated from within the build pod
 func (detailsStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newBuild := obj.(*api.Build)
 	oldBuild := old.(*api.Build)
 	revision := newBuild.Spec.Revision
+	message := newBuild.Status.Message
+	reason := newBuild.Status.Reason
 	*newBuild = *oldBuild
 	newBuild.Spec.Revision = revision
+	newBuild.Status.Reason = reason
+	newBuild.Status.Message = message
 }
 
 // Validates that an update is valid by ensuring that no Revision exists and that it's not getting updated to blank
 func (detailsStrategy) ValidateUpdate(ctx kapi.Context, obj, old runtime.Object) field.ErrorList {
 	newBuild := obj.(*api.Build)
 	oldBuild := old.(*api.Build)
+	oldRevision := oldBuild.Spec.Revision
+	newRevision := newBuild.Spec.Revision
+	oldReason := oldBuild.Status.Reason
+	oldMessage := oldBuild.Status.Message
+	newReason := newBuild.Status.Reason
+	newMessage := newBuild.Status.Message
 	errors := field.ErrorList{}
-	if oldBuild.Spec.Revision != nil {
+
+	if newRevision == nil {
+		errors = append(errors, field.Invalid(field.NewPath("spec", "revision"), nil, "cannot set an empty revision in build spec"))
+	}
+	// NOTE: use reflect.DeepEqual to see if the old one is as the new one? and
+	// skip this step that causes issues with updating other fields on the build
+	if !reflect.DeepEqual(oldRevision, newRevision) && oldRevision != nil {
 		// If there was already a revision, then return an error
-		errors = append(errors, field.Duplicate(field.NewPath("status", "revision"), oldBuild.Spec.Revision))
+		errors = append(errors, field.Duplicate(field.NewPath("spec", "revision"), oldBuild.Spec.Revision))
+	}
+
+	// NOTE: We don't want to update the reason and message to nothing?
+	if len(oldReason) > 0 && len(newReason) == 0 {
```

Comment:
i don't think you need either of these checks.  i think you always want to set reason and message to whatever value the updated build has.